src/web/Popup.ts

```diff
@@ -12,16 +12,27 @@ import RX = require('../common/Interfaces');
 import Types = require('../common/Types');
 
 export class Popup extends RX.Popup {
-
     show(options: Types.PopupOptions, popupId: string, delay?: number): boolean {
+        if (!popupId || popupId === '') {
```

Comment:
Maybe would be better move this logic to separate method? And will be one method with the same logic. What do you think, does it make sense?

```
private isEmpty(popupId: string) {
   if (!popupId || popupId === '') {
      throw new Error(`popupId must be a non-empty string. Actual: ${popupId}`);
   }
}
```

Reply:
In this use case (parameter validation), I think the better stack trace is worth the small code duplication.

If we were to start extracting these into separate functions, then I'd advocate for introducing a lib to handle that consistently across the project, e.g. [validate](https://github.com/mdvorscak/validate-params), [check-preconditions](https://github.com/pnann/check-preconditions)